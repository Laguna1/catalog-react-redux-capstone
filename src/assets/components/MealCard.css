.meal-card-wrapper {
  width: 18vw;
  min-width: 20vh;
  margin-bottom: 20px;
  background-color: #fdfbfa;
  box-shadow: 1px 5px 9px 2px #d1d1d1;
  transition: box-shadow 0.5s, border 1s;
}

.meal-card {
  padding-bottom: 2.5em;
}

.meal-card-image {
  width: 90%;
  height: 180px;
  object-fit: cover;
}

.meal-card-link {
  font-weight: 700;
  font-size: small;
  color: black;
}
