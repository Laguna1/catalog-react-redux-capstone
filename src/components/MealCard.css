.meal-card-wrapper {
  width: 24vw;
  min-width: 42vh;
  margin-bottom: 20px;
  background-color: #fdfbfa;
  box-shadow: 1px 5px 9px 2px #d1d1d1;
  transition: box-shadow 0.5s, border 1s;
}

.meal-card {
  border: 2px solid green;
  padding-bottom: 3em;
}

.meal-card-image {
  width: 85%;
  height: 250px;
  object-fit: cover;
}
