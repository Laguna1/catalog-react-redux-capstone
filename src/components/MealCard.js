import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../assets/components/MealCard.css';

const MealCard = ({ meal, clickHandler }) => {
  const {
    idMeal,
    strMeal,
    strMealThumb,
  } = meal;

  const handleClick = meal => {
    clickHandler(meal);
  };

  return (
    <div className="meal-card-wrapper">
      <div className="meal-card p-3" key={idMeal}>
        <img src={strMealThumb} alt={strMeal} className="meal-card-image my-3" />
        <div className="title">
          <Link
            to={{
              pathname: `/meals/${idMeal}`,
              state: { meal },
            }}
            onClick={() => handleClick(meal)}
            className="meal-card-link"
          >
            {meal.strMeal}
            mealcard
          </Link>
        </div>
      </div>
    </div>
  );
};

MealCard.propTypes = {
  meal: PropTypes.shape({
    idMeal: PropTypes.string.isRequired,
    strMeal: PropTypes.string.isRequired,
    strMealThumb: PropTypes.string,
  }).isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default MealCard;
