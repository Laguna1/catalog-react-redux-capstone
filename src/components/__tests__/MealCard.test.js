import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MealCard from '../MealCard';

const mealTest = {
  idMeal: '52525',
  strMeal: 'My meal',
  strMealThumb: 'https://www.google.com/images/test.png',
};

const renderer = new ShallowRenderer();

it('should render MealCard correctly 1', () => {
  renderer.render(<MealCard meal={mealTest} />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
