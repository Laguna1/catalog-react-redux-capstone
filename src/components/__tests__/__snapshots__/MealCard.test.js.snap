// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MealCard correctly 1`] = `
<div
  className="meal-card-wrapper"
>
  <div
    className="meal-card p-3"
  >
    <img
      alt="My meal"
      className="meal-card-image my-3"
      src="https://www.google.com/images/test.png"
    />
    <div
      className="title"
    >
      <Link
        className="meal-card-link text-decoration-none"
        onClick={[Function]}
        to={
          Object {
            "pathname": "/meals/52525",
            "state": Object {
              "meal": Object {
                "idMeal": "52525",
                "strMeal": "My meal",
                "strMealThumb": "https://www.google.com/images/test.png",
              },
            },
          }
        }
      >
        My meal
      </Link>
    </div>
  </div>
</div>
`;
